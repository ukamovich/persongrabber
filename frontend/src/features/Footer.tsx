import React from "react";
import './personGrabber/styles/footer.css'
import { useAppSelector } from '../hooks'

/**
 * Footer containing clicked people history
 * @returns 
 */
function Footer() {

    const history = useAppSelector((state) => state.visited)
    var columns = () => {
        switch (history.length) {
            case 1:
                return 1
            case 2:
                return 2
            default:
                return (history.length < 7) ? 3 : 4
        }
    }

    return (
        <footer className="footer" >
            <h4>Clicked people history:</h4>
            <ul style={{columns: columns(), listStyleType: "none"}}>
                {history && history.map((el, index) => {
                    return <li key={`Person: ${index}`}>{el}</li>
                })}
            </ul>
        </footer>
    );

}
export default Footer
