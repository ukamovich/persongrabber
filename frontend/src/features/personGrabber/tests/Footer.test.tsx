import Footer from "../../Footer";
import renderer from "react-test-renderer";
import { BrowserRouter } from "react-router-dom";
import { Provider } from "react-redux";
import store from "../../../store";

// Snapshot testing with Jest
describe("Footer", () => {
  it("renders properly", () => {
    const treeFooter = renderer
      .create(
        <BrowserRouter>
          <Provider store={store}>
            <Footer />
          </Provider>
        </BrowserRouter>
      )
      .toJSON();
    expect(treeFooter).toMatchSnapshot();
  });
});
