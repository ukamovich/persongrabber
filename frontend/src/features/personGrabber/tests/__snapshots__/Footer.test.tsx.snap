// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders properly 1`] = `
<footer
  className="footer"
>
  <h4>
    Clicked people history:
  </h4>
  <ul
    style={
      Object {
        "columns": 3,
        "listStyleType": "none",
      }
    }
  />
</footer>
`;
